src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -360,22 +386,32 @@ def step(carry):
                     A=A,
                     b=b,
                 )
+
                 params = optax.apply_updates(params, updates)
+
                 return params, state
 
-            def continuing_criterion(carry):
+            def continuing_criterion(carry: OptimizationCarry) -> jtp.Bool:
+
                 _, state = carry
+
                 iter_num = optax.tree_utils.tree_get(state, "count")
                 grad = optax.tree_utils.tree_get(state, "grad")
                 err = optax.tree_utils.tree_l2_norm(grad)
+
                 return (iter_num == 0) | ((iter_num < maxiter) & (err >= tol))
 
-            init_carry = (init_params, opt.init(init_params))
             final_params, final_state = jax.lax.while_loop(
                 continuing_criterion, step, init_carry
             )
+
             return final_params, final_state
 
+        # ======================================
+        # Compute the contact forces with L-BFGS
+        # ======================================
+
+        # Initialize the optimized forces with a linear Hunt/Crossley model.
         init_params = (
             K[:, jnp.newaxis] * jnp.zeros_like(position).at[:, 2].set(δ)
             + D[:, jnp.newaxis] * velocity
```

Comment:
At this point we can use the normal penetration velocity
```suggestion
            + D[:, jnp.newaxis] * jnp.zeros_like(position).at[:, 2].set(δ_dot)
```

Reply:
Same as above. I noticed that this contact model is very sensitive to this warm start. It's something worth investigating.